packages/api-explorer/src/components/SideNav/SideNav.tsx

```diff
@@ -84,12 +84,12 @@ export const SideNav: FC<SideNavProps> = ({ headless = false, spec }) => {
     if (parts[1] === 'diff') {
       if (parts[3] !== tabNames[index]) {
         parts[3] = tabNames[index]
-        navigateWithGlobalParams(parts.join('/'))
+        navigate(parts.join('/'), { t: null })
```

Comment:
The reason for this change: if we are in a diff scene when we have `opts` in the url, and we open the SideNav to switch to the `Types` or `Method` tab, the diff scene shouldn't change - we should maintain the current parameter as is instead of resetting it. So in this case, I feel it's appropriate that navigation should specifically remove the tag filter variable only.